Migrate Frontend from CRA to Vite and Integrate Tailwind CSS v4

Ramharsh-aidev opened the pull request:
# issue #16

### This PR completes the migration of the RECIPEDIA frontend from **Create React App (CRA)** to **Vite**, and integrates **Tailwind CSS v4** for modern styling and improved developer experience.

#### 🔧 Key Changes
- Replaced CRA with Vite for faster builds and better compatibility with Tailwind
- Installed and configured Tailwind CSS v4 using Vite plugin
- Updated `package.json` scripts (`npm run dev`, `npm run build`, `npm run preview`)
- Refactored file structure to align with Vite conventions
- Replaced `react-scripts` with Vite dependencies
- Updated `index.html` and entry points for Vite
- Renamed `App.js` to `App.jsx` for consistency
- Updated `index.css` with Tailwind directives (`@tailwind base`, `components`, `utilities`)
- Modified `README.md` to reflect new setup and run instructions

#### 🎯 Benefits
- ⚡ Faster development and build times
- 🎨 Cleaner, utility-first styling with Tailwind
- 🧩 Easier future integration of animations and responsive design
- 📖 Clearer documentation for contributors

#### ✅ Acceptance Criteria
- App runs successfully using Vite (`npm run dev`)
- Tailwind styles are applied correctly
- README instructions match the new setup
- No CRA-specific configs or scripts remain

Arishsingh commented:
You're absolutely right @Ramharsh-aidev  about Tailwind CSS v4 — with Vite, the postcss.config.js file is no longer required by default if you're using the plugin directly in vite.config.js, which you've correctly done. Thanks for pointing that out and following the official docs.
Everything looks good now. Great work on this migration — well-structured and thoughtfully done. Approved and ready to merge! 
— @Arishsingh (Mentor, GSSoC'25)

Ramharsh-aidev commented:
ok as this is not UI update, I will share that, some changes which I made.
<br> here, its running using command npm run dev instead of the npm start 

<img width="877" height="308" alt="image" src="https://github.com/user-attachments/assets/56fa3a6a-4d40-47ed-943c-72649c695751" />

below is screenshot of package.json where I changed the scripts.

<img width="532" height="295" alt="image" src="https://github.com/user-attachments/assets/ac69fe4d-5e6f-4e7b-9ec3-93193e4088b7" />

I had changed the file names of App.js to App.jsx and index.js to index.jsx and also changed the position of the index.html

<img width="498" height="365" alt="image" src="https://github.com/user-attachments/assets/47ddb659-2771-4842-8278-e7de78e9515b" />

and here is the plugin of tailwind added
<img width="781" height="383" alt="image" src="https://github.com/user-attachments/assets/b361545c-a96e-48d2-a973-edd31bbfcb53" />


I hope this is sufficient. If you need something else to verify, please tell me what you need specifically.

Arishsingh commented:
Hi @Ramharsh-aidev  ,
It looks like your PR has merge conflicts —  Could you please resolve the conflicts on your end? If you're facing any issues or something seems off, feel free to share a screenshot so we can review it better and assist you accordingly.
Thanks for your contribution!

Ramharsh-aidev commented:
Yes I will try to resolve conflicts from my side

Arishsingh commented:
okay @Ramharsh-aidev  i will do something for that
